java-checks/src/main/java/org/sonar/java/checks/ConstantsShouldBeStaticFinalCheck.java

```diff
@@ -96,7 +96,20 @@ private static boolean hasConstantInitializer(VariableTree variableTree) {
       if (init.is(Tree.Kind.NEW_ARRAY)) {
         return false;
       }
-      return !containsChildrenOfKind((JavaTree) init, Tree.Kind.METHOD_INVOCATION, Tree.Kind.NEW_CLASS);
+      return !containsChildMatchingPredicate((JavaTree) init,
+        ((Predicate<Tree>) ConstantsShouldBeStaticFinalCheck::isIgnoredKind).or(ConstantsShouldBeStaticFinalCheck::isThisOrSuper));
```

Comment:
Looks like my suggestion was not that elegant. For me it will be more readable to use lambda. 
```suggestion
        (tree -> isIgnoredKind(tree) || isThisOrSuper(tree)));
```

Reply:
Of course, thanks!